com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HDShaderUtils.cs

```diff
@@ -105,10 +105,14 @@ public static bool ResetMaterialKeywords(Material material)
 
             if (resetter != null)
             {
+                if (HDSpeedTree8MaterialUpgrader.IsHDSpeedTree8Material(material))
+                    HDSpeedTree8MaterialUpgrader.SaveHDSpeedTree8Setup(material);
                 CoreEditorUtils.RemoveMaterialKeywords(material);
                 // We need to reapply ToggleOff/Toggle keyword after reset via ApplyMaterialPropertyDrawers
                 MaterialEditor.ApplyMaterialPropertyDrawers(material);
                 resetter(material);
+                if (HDSpeedTree8MaterialUpgrader.IsHDSpeedTree8Material(material))
+                    HDSpeedTree8MaterialUpgrader.RestoreHDSpeedTree8Setup(material);
```

Comment:
I am not sure you need to do that. Looking at the save/restore code, I don't see why you need to save and restore _WINDQUALITY and _CullMode, which are float properties so they don't get resetted.
About the EFFECT_BILLBOARD, it should be set or unset from inside the resetter function (the one of sg lit in your case) depending on the value of the associated property (which i don't find by the way, in the previous PR there was some code based on the name of the material but not anymore)

Reply:
They get overwritten to default values- either during SynchronizeShaderGraphProperties as part of resetting, or during ApplyMaterialPropertyDrawers, since the material has been initialized by the importer, but the drawer has not. 

Let me move this code inside the resetter to see if the correct wind property gets restored after ApplyMaterialPropertyDrawers.

Reply:
Yeah unfortunately ApplyMaterialPropertyDrawers resets windquality and cullmode. Is there a way to update the materialpropertydrawer so that this doesn't happen?

EFFECT_BILLBOARD would be the name of the float toggling the EFFECT_BILLBOARD keyword and would have the same problem as wind and cullmode, where it gets reset by ApplyMaterialPropertyDrawers. I can still do it by the name of the material but that feels hacky and prevents users from disabling it (which becomes relevantin later releases of Graphics, where you can override builtin keywords through the SG UI).

Reply:
ApplyMaterialPropertyDrawers does not modify the property values, it only sets keywords based on the value. So it's probably something else that modifies the values. But I checked out the branch to test and for me the values were not resetted so maybe i was doing something wrong.

For EFFECT_BILLBOARD, what i was trying to say is that ther's no way (or i haven't found it) to modify the keyword value from the UI, but I believe there should be a property and the keywoard should be set depending on the property value (the keyword would actually be set by ApplyMaterialPropertyDrawers)

Reply:
I confirmed that the floats seem to stick now. I must have been resetting them myself as I iterated on changes without fully understanding the system. :) Thanks so much for the help.

Cleaned up here: https://github.com/Unity-Technologies/Graphics/pull/3789/commits/6893f6902e91500e5f5b7ab3969d3b9f4d073edc